dds/DCPS/ReceivedDataElementList.h

```diff
@@ -51,38 +52,48 @@ class OpenDDS_Dcps_Export ReceivedDataElement {
       ref_count_(1),
       mx_(mx)
   {
-
     this->destination_timestamp_ = time_value_to_time(ACE_OS::gettimeofday());
 
     this->source_timestamp_.sec = header.source_timestamp_sec_;
     this->source_timestamp_.nanosec = header.source_timestamp_nanosec_;
+
+    // When we have a dispose or unregister instance we shouldn't let the
```

Comment:
This could be reworded in terms of SampleInfo from the spec (which is where `valid_data` comes from)